src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/ProvideCorrectArgumentsToFormattingMethodsTests.cs

```diff
@@ -371,7 +850,18 @@ class Test
 
     void M1(string param)
     {
-        var a = MyFormat("""", 1);
+        // Too many args
+        var s1 = MyFormat(""{0}"", 1, 2);
+        // Too many args and missing format index
+        var s2 = MyFormat(""{1}"", 1, 2, 3);
+
+        // Not enough args
+        var s3 = MyFormat(""{0} {1}"", 1);
+        // Not enough args and missing format index
+        var s4 = MyFormat(""{0} {2}"", 1);
```

Comment:
I am tempted to have a separate set of descriptors for these cases (they will use CA2241 as the too many  cases) because we are not sure these will lead to runtime failure as this is a custom implementation.

Reply:
I would be fine with not issuing these diagnostics at all to start. Yes, historically we've done this but and I can see the utility but I would not turn these on for folks by default. This is always going to be something that you need to opt into. I like the idea of having a separate ID for custom methods so folks can still get these warnings on their custom loggers.

---
In reply to: [508699816](https://github.com/dotnet/roslyn-analyzers/pull/4226#discussion_r508699816) [](ancestors = 508699816)